pymongo/asynchronous/mongo_client.py

```diff
@@ -1173,6 +1175,22 @@ def __getitem__(self, name: str) -> database.AsyncDatabase[_DocumentType]:
         """
         return database.AsyncDatabase(self, name)
 
+    def __del__(self) -> None:
+        """Check that this AsyncMongoClient has been closed and issue a warning if not."""
+        # TODO: Remove in https://jira.mongodb.org/browse/PYTHON-4731
+        try:
+            if not self._closed:
+                warnings.warn(
+                    f"Unclosed {self}",
```

Comment:
We should add more information here so users know why this started.

Reply:
What kind of information should be here? Something like `Unclosed MongoClient. Call MongoClient.close() to safely shut down your client and free up resources.`?

Reply:
Do we want to pull that ticket up to 4.9, then? We should do them both in the same release.